### PR TITLE
Fix data preprocessing for chatbot tutorial

### PR DESCRIPTION
**Pull request by priyaramani**
Location and format of cornell movie-dialogs corpus have changed a bit (format changed from 2 text files-movie_lines and movie_conversations to a single utterances json file) and so the existing tutorial doesn't load and process the new data format properly. Updated the code and comments to work on the updated data format.

Working of updated code
```
(venv38) ?130 priyaramani@priyaramani-mbp chatbot % python chatbot.py

b'{"id": "L1045", "conversation_id": "L1044", "text": "They do not!", "speaker": "u0", "meta": {"movie_id": "m0", "parsed": [{"rt": 1, "toks": [{"tok": "They", "tag": "PRP", "dep": "nsubj", "up": 1, "dn": []}, {"tok": "do", "tag": "VBP", "dep": "ROOT", "dn": [0, 2, 3]}, {"tok": "not", "tag": "RB", "dep": "neg", "up": 1, "dn": []}, {"tok": "!", "tag": ".", "dep": "punct", "up": 1, "dn": []}]}]}, "reply-to": "L1044", "timestamp": null, "vectors": []}\n'
b'{"id": "L1044", "conversation_id": "L1044", "text": "They do to!", "speaker": "u2", "meta": {"movie_id": "m0", "parsed": [{"rt": 1, "toks": [{"tok": "They", "tag": "PRP", "dep": "nsubj", "up": 1, "dn": []}, {"tok": "do", "tag": "VBP", "dep": "ROOT", "dn": [0, 2, 3]}, {"tok": "to", "tag": "TO", "dep": "dobj", "up": 1, "dn": []}, {"tok": "!", "tag": ".", "dep": "punct", "up": 1, "dn": []}]}]}, "reply-to": null, "timestamp": null, "vectors": []}\n'
b'{"id": "L985", "conversation_id": "L984", "text": "I hope so.", "speaker": "u0", "meta": {"movie_id": "m0", "parsed": [{"rt": 1, "toks": [{"tok": "I", "tag": "PRP", "dep": "nsubj", "up": 1, "dn": []}, {"tok": "hope", "tag": "VBP", "dep": "ROOT", "dn": [0, 2, 3]}, {"tok": "so", "tag": "RB", "dep": "advmod", "up": 1, "dn": []}, {"tok": ".", "tag": ".", "dep": "punct", "up": 1, "dn": []}]}]}, "reply-to": "L984", "timestamp": null, "vectors": []}\n'
b'{"id": "L984", "conversation_id": "L984", "text": "She okay?", "speaker": "u2", "meta": {"movie_id": "m0", "parsed": [{"rt": 1, "toks": [{"tok": "She", "tag": "PRP", "dep": "nsubj", "up": 1, "dn": []}, {"tok": "okay", "tag": "RB", "dep": "ROOT", "dn": [0, 2]}, {"tok": "?", "tag": ".", "dep": "punct", "up": 1, "dn": []}]}]}, "reply-to": null, "timestamp": null, "vectors": []}\n'
.
.
.


Processing corpus into lines and conversations...

Writing newly formatted file...

Sample lines from file:
b'They do to!\tThey do not!\n'
b'She okay?\tI hope so.\n'
b"Wow\tLet's go.\n"
b'"I\'m kidding.  You know how sometimes you just become this ""persona""?  And you don\'t know how to quit?"\tNo\n'
b"No\tOkay -- you're gonna need to learn how to lie.\n"
b"I figured you'd get to the good stuff eventually.\tWhat good stuff?\n"
b'What good stuff?\t"The ""real you""."\n'
b'"The ""real you""."\tLike my fear of wearing pastels?\n'
b'do you listen to this crap?\tWhat crap?\n'
b"What crap?\tMe.  This endless ...blonde babble. I'm like, boring myself.\n"
Start preparing training data ...
Reading lines...
Read 221282 sentence pairs
Trimmed to 64313 sentence pairs
Counting words...
Counted words: 18082

pairs:
['they do to !', 'they do not !']
['she okay ?', 'i hope so .']
['wow', 'let s go .']
['what good stuff ?', 'the real you .']
['the real you .', 'like my fear of wearing pastels ?']
['do you listen to this crap ?', 'what crap ?']
['well no . . .', 'then that s all you had to say .']
['then that s all you had to say .', 'but']
['but', 'you always been this selfish ?']
['have fun tonight ?', 'tons']
keep_words 7833 / 18079 = 0.4333
Trimmed from 64313 pairs to 53131, 0.8261 of total
input_variable: tensor([[  34, 1681,   86,   40, 3102],
        [ 294,   35,   14,   24,   14],
        [2423,    8,   14,  208,    2],
        [  10,   17,   14,  135,    0],
        [3659,  350,   22,   10,    0],
        [7624, 1507, 6074,    2,    0],
        [  10,   14,   14,    0,    0],
        [   2,    2,    2,    0,    0]])
lengths: tensor([8, 8, 8, 6, 3])
target_variable: tensor([[  34,    8,  128,   34, 2412],
        [   3,   17, 4537,   11, 2306],
        [ 246,  321,    6,  419,   14],
        [ 135,   14,    2,   85,    2],
        [  39,  575,    0,    5,    0],
        [   2,    2,    0,  548,    0],
        [   0,    0,    0,  268,    0],
        [   0,    0,    0,  411,    0],
        [   0,    0,    0,   14,    0],
        [   0,    0,    0,    2,    0]])
mask: tensor([[ True,  True,  True,  True,  True],
        [ True,  True,  True,  True,  True],
        [ True,  True,  True,  True,  True],
        [ True,  True,  True,  True,  True],
        [ True,  True, False,  True, False],
        [ True,  True, False,  True, False],
        [False, False, False,  True, False],
        [False, False, False,  True, False],
        [False, False, False,  True, False],
        [False, False, False,  True, False]])
max_target_len: 10
Building encoder and decoder ...
Models built and ready to go!
Building optimizers ...
Starting Training!
Initializing ...
Training...
Iteration: 1; Percent complete: 0.0%; Average loss: 8.9759
Iteration: 2; Percent complete: 0.1%; Average loss: 8.8593
.
.
```

### CODE REVIEW

**Review by malfet — CHANGES_REQUESTED** [review]
Well, it would not work until we re-upload the data to s3

File: Makefile [hunk 1]
```diff
@@ -73,6 +73,7 @@ download:
 	wget -N https://s3.amazonaws.com/pytorch-tutorial-assets/iris.data -P $(DATADIR)
 	cp $(DATADIR)/iris.data beginner_source/data/
 
+	# TODO Download updated movie-corpus zip file, upload it to pytorch-tutorial-assets and use that link here
 	# Download dataset for beginner_source/chatbot_tutorial.py
 	wget -N https://s3.amazonaws.com/pytorch-tutorial-assets/cornell_movie_dialogs_corpus.zip -P $(DATADIR)
 	unzip $(ZIPOPTS) $(DATADIR)/cornell_movie_dialogs_corpus.zip -d beginner_source/data/
```
Comment (malfet) [inline Makefile:79]:
```suggestion
	wget -N https://s3.amazonaws.com/pytorch-tutorial-assets/cornell_movie_dialogs_corpus_v2.zip -P $(DATADIR)
	unzip $(ZIPOPTS) $(DATADIR)/cornell_movie_dialogs_corpus_v2.zip -d beginner_source/data/
```

File: Makefile [hunk 2]
```diff
@@ -73,6 +73,7 @@ download:
 	wget -N https://s3.amazonaws.com/pytorch-tutorial-assets/iris.data -P $(DATADIR)
 	cp $(DATADIR)/iris.data beginner_source/data/
 
+	# TODO Download updated movie-corpus zip file, upload it to pytorch-tutorial-assets and use that link here
```
Comment (malfet) [inline Makefile:76]:
```suggestion
```

**Review by malfet — APPROVED** [review]
LGTM provided that CI is green

**Comment by priyaramani** [comment]
Thanks @malfet for helping upload the updated data set!

**Pull request merged**